src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    Animal.alive = Animal.alive.filter((animal) => animal !== this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (!(herbivore instanceof Herbivore) || herbivore.hidden) {
+      return;
+    }
+
+    herbivore.health -= 50;
+
+    if (herbivore.health <= 0) {
+      herbivore.die();
+    }
+  }
```

Comment:
[CODE PERFORMANCE] - Instead of checking the health of the herbivore after each bite and calling `die` method, consider filtering the `Animal.alive` array at the end of the `bite` method to remove any animals with health less than or equal to 0. This will avoid the performance overhead of updating the array multiple times if multiple bites occur in a short time span.